src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +19,78 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            this.upperBound= Integer.parseInt(argument);
+
+        } catch (NumberFormatException e) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new IllegalArgumentException(UsefulConstants.WRONG_ARGUMENT);
+        }
+
+        if (this.upperBound < 1 || this.upperBound > UsefulConstants.MAX_UPPER_BOUND) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new IllegalArgumentException(UsefulConstants.WRONG_ARGUMENT);
+        }
+
+        this.maxAttempts=(int)(Math.log(this.upperBound) / Math.log(2))+1;
+        this.numberToGuess=(int)(Math.random()*upperBound)+1;
+    }
+
+    private void printProgressBar(int a, int b) {
+
+        System.out.print("Twoje próby: [");
+
+        for(int i=0;i<a;i++) {
+            System.out.print("*");
+        }
+
+        for(int i=0;i<b;i++) {
+            System.out.print(".");
+        }
+
```

Comment:
_clean code_ Możnaby tutaj skorzystać z:
https://www.geeksforgeeks.org/string-class-repeat-method-in-java-with-examples/